File: Onechemic/Core/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class Product
    {
        public int Id { get; set; }

        public string TitleEn { get; set; } = string.Empty;
        public string TitleAr { get; set; } = string.Empty;

        public string DescriptionEn { get; set; } = string.Empty;
        public string DescriptionAr { get; set; } = string.Empty;

        public string DetailedDescriptionEn { get; set; } = string.Empty;
        public string DetailedDescriptionAr { get; set; } = string.Empty;

        public string ImagePath { get; set; } = string.Empty;
        public string Nav { get; set; } = string.Empty;

        public string CategoryEn { get; set; } = string.Empty;
        public string CategoryAr { get; set; } = string.Empty;
    }
}

File: Frontend/admin/types/product.ts
// Product type matching backend API response (camelCase from ASP.NET Core)
export type Product = {
  id: number;
  titleEn: string;
  titleAr: string;
  descriptionEn: string;
  descriptionAr: string;
  detailedDescriptionEn: string;
  detailedDescriptionAr: string;
  imagePath: string;
  nav: string;
  categoryEn: string;
  categoryAr: string;
};

export type Category = {
  en: string;
  ar: string;
};

export type ModalType = "add" | "edit" | "view" | "delete" | null;

export type FormData = {
  titleEn: string;
  titleAr: string;
  descriptionEn: string;
  descriptionAr: string;
  detailedDescriptionEn: string;
  detailedDescriptionAr: string;
  categoryEn: string;
  categoryAr: string;
  nav: string;
};
